src/PlusDataCollection/MicrosoftMediaFoundation/vtkPlusMmfVideoSource.cxx

```diff
@@ -609,6 +611,16 @@ PlusStatus vtkPlusMmfVideoSource::AddFrame(unsigned char* bufferData, DWORD buff
     return PLUS_SUCCESS;
   }
 
+  double targetSecondsBetweenFrames = 1.0 / this->GetAcquisitionRate();
+  double secondsSinceLastFrame = vtkIGSIOAccurateTimer::GetSystemTime() - this->LastFrameTime;
+  if (this->LastFrameTime > 0 && secondsSinceLastFrame < targetSecondsBetweenFrames)
+  {
+    // The required time has not elapsed between frames.

```

Comment:
explain why this can happen for this video source (i.e., you don't always get what you request)